.github/workflows/shared-build-deploy-container.yml

```diff
@@ -0,0 +1,140 @@
+        description: "Personal access token used to access the github private nuget source."
+        required: true
+    inputs:
+      runnerName:
+        description: "Name of the runner to use."
+        required: true
+        type: string
+      projectFolder:
+        description: "Folder containing the project to build."
+        required: true
+        type: string
+      imageName:
+        description: "Name of the docker image to build."
+        required: true
+        type: string
+      serviceName:
+        required: true
+        type: string
+      environment:
+        description: "Environment to deploy to (dev | staging | production)."
+        required: true
+        type: string
+      containerPublishType:
+        description: "Type of container publish (api | worker)."
+        required: true
+        type: string
+      imageTagPrefix:
+        description: "Prefix of the docker image tag. If not provided, will be generated based on the environment."
+        required: false
+        type: string
+        default: ""
+      dotnetVersion:
+        description: "Version of dotnet to use. Default is v7.x."
+        required: false
+        type: string
+        default: "7.x"
+      dockerRegistry:
+        description: "Docker registry to push the image to."
+        required: false
+        type: string
+        default: "docker.pkg.github.com"
+      actionsRepoRef:
+        description: "Run actions from this ref. Default is master."
+        type: string
+        required: false
+        default: "master"
+
+jobs:
+  build-push-container:
+    name: "Build and push container image"
+    runs-on: ${{ inputs.runnerName }}
+    steps:
+      - name: Checkout calling repo
+        uses: actions/checkout@v3
+
+      - name: Checkout trakx/github-actions repo
+        uses: actions/checkout@v3
+        with:
+          repository: trakx/github-actions
+          path: ./github-actions
+          ref: ${{ inputs.actionsRepoRef }}
+
+      - name: Get last commit of calling repo
+        id: last-commit-sha
+        uses: ./github-actions/get-last-commit
+
+      - name: Set variable CONTAINER_IMAGE_TAG_PREFIX
+        shell: bash
+        run: |
+          IMAGE_TAG="${{inputs.imageTagPrefix}}"
+          if [[ -z "$IMAGE_TAG" ]]; then
+            if [[ "${{inputs.environment}}" == "production" ]]; then
+                IMAGE_TAG="prod"
+            elif [[ "${{inputs.environment}}" == "staging" ]]; then
+                IMAGE_TAG="stage"
+            else
+                IMAGE_TAG="dev"
+            fi
+          fi
+          echo "Tagging image prefix with: $IMAGE_TAG"
+          echo "CONTAINER_IMAGE_TAG_PREFIX=$IMAGE_TAG" >> $GITHUB_ENV
+
+      - name: Build and push container image
+        id: build-push-container-step
+        uses: ./github-actions/build-push-container
+        with:
+          githubToken: ${{ secrets.TRAKX_GITHUB_TOKEN }}
+          packageReadonlyPat: ${{secrets.TRAKX_BOT_READONLY_PAT}}
```

Comment:
I can't remember why we use this instead of giving the GitHubTokwn with appropriate permissions 🤔

Reply:
I don't know either 🤣 
I'll open a task to check this later.